dev/reference/garbage-collection/overview.md

```diff
@@ -0,0 +1,45 @@
+---
+title: GC Overview
+summary: Overview of Garbage Collection in TiDB.
+category: reference
+---
+
+# GC Overview
+
+TiDB uses MVCC to control transaction concurrency. When you update the data, the original data is not deleted immediately but is kept together with the new data, with a timestamp to distinguish the version. The goal of Garbage Collection (GC) is to clear the obsolete data.
+
+## GC process
+
+Each TiDB cluster contains a TiDB instance that is selected as the GC leader, which controls the GC process.
+
+GC runs periodically on TiDB. The default frequency is once every 10 minutes. For each GC, TiDB firstly calculates a timestamp called "safe point" (defaults to the current time minus 10 minutes). Then, TiDB clears the obsolete data under the premise that all the snapshots after the safe point retain the integrity of the data. Specifically, there are three steps involved in the GC process:
+
+1. Resolve Locks
+2. Delete Ranges
+3. Do GC
+
+### Resolve Locks
+
+The TiDB transaction model is implemented based on [Google's Percolator](https://ai.google/research/pubs/pub36726). It's mainly a two-phase commit protocol with some practical optimizations. When the first phase is finished, all the related keys are locked. Among these locks, one is the primary lock and the others are secondary locks which contain a pointer to the primary lock; in the secondary phase, the key with the primary lock gets a write record and its lock is removed. The write record indicates the write or delete operation in the history or the transactional rollback record of this key. The type of write record that replaces the primary lock indicates whether the corresponding transaction is committed successfully. Then all the secondary locks are replaced successively. If the threads fail to replace the secondary locks, these locks are retained.
+
+The Resolve Locks step rolls back or commits the locks before the safe point, depending on whether their primary key has been committed or not. If the primary key is also retained, the transaction times out and will be rolled back.
+
+> **Note:**
+>
+> Resolve Locks is a required step. Once GC has cleared the write record of the primary lock, you can never know whether this transaction is successful or not. Also, if the the transaction contains retained secondary keys, it's important to know whether it should be rolled back or committed. As a result, data consistency can't be guaranteed.
```

Comment:
This is not consistent with the Chinese version, which is not written in a "Note" block.

Reply:
I think this looks like a Note material when I translated it so I took the liberty to make it one.  I will update the Chinese version in a separate PR.